controller/src/main/java/org/jboss/as/controller/operations/global/ReadAttributeHandler.java

```diff
@@ -281,9 +281,19 @@ static void resolveAttribute(OperationContext context, AttributeDefinition attri
             final ModelNode result = subModel.get(attribute.getName());
             context.getResult().set(result);
         } else if (defaults && attribute.getDefaultValue() != null) {
-            // No defined value in the model. See if we should reply with a default from the metadata,
-            // reply with undefined, or fail because it's a non-existent attribute name
-            context.getResult().set(attribute.getDefaultValue());
+            if(!attribute.hasAlternative(subModel)) {
```

Comment:
I was backporting this change to EAP 7.0 and I've found that there is a flag, that attribute is allowed to be null. I think that even in case that there is an alternative, but field cannot be null, we can not set it to null. I backported it with this change - https://github.com/jbossas/wildfly-core-eap/pull/378 (only difference is on this line) - may I ask for opinion?